### PR TITLE
Feat/training, train.py

### PR DESCRIPTION
**Pull request by lsig**
Code for epoch training has been set up, type annotated.

### CODE REVIEW

File: training/train.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import torch
+from torch.optim import AdamW
+from torch.optim.lr_scheduler import LambdaLR
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import PreTrainedModel
+
+device = torch.device("cpu:0" if torch.cuda.is_available() else "cpu")
+
+def set_model(model: PreTrainedModel, batch: DataLoader[tuple[torch.Tensor, torch.Tensor]]) -> PreTrainedModel:
```
Comment (njallskarp) [inline training/train.py:10]: Would rename this to 'get_prediction'

**Comment by njallskarp** [comment]
Added one comment, otherwise looks good

**Review by eysta00 — APPROVED** [review]
LGTM! Although I am no professional.

**Review by njallskarp — CHANGES_REQUESTED** [review]
added one comment. please fix

File: training/train.py (hunk 1 above)
Comment (lsig) [reply training/train.py:10]: Will do

Comment (njallskarp) [reply training/train.py]: noice

**Review by njallskarp — APPROVED** [review]

**Pull request merged**